src/axom/primal/operators/detail/intersect_bezier_impl.hpp

```diff
@@ -468,10 +471,31 @@ bool intersect_2d_circle_line(const Sphere<T, 2> &circ,
 
   T disc = circ.getRadius() * circ.getRadius() * dr * dr - D * D;
 
-  // Treat tangencies as *not* intersecting
-  if(disc <= 0.0)
+  // Identify near-tangent cases
+  if(disc <= EPS * EPS)
   {
-    return false;
+    T ct;
+    Segment<T, 2> seg(a, b);
+
+    // Because the line is known to be tangent, there's only one intersection point,
+    //  and it would have to be at the closest point on the line to the circle center.
+    Point<T, 2> cp = closest_point(circ.getCenter(), seg, &ct);
+    T dist = primal::squared_distance(cp, circ.getCenter());
+
+    if(axom::utilities::isNearlyEqual(dist, circ.getRadius(), EPS))
```

Comment:
Did you try this where T=float? I ask because I ran into problems with ``isNearlyEqual()`` in polyhedron where T was float and the RPS is double. It might be necessary to cast EPS.

Reply:
Good catch! That does cause a problem here, and probably in about half a dozen other places. I don't think there's a reason that EPS _shouldn't_ be float, so I think another solution would just to make sure those tolerance variables are templated same as everything else.

Reply:
After my first attempt to fix this robustly sent me down a bit of a rabbit hole, I think to fully untangle this across primal (particularly in my winding number and intersection methods, I'll admit) probably merits more consideration than an isolated fix in an unrelated PR.

I've created an issue #1563 to track this, and have put my efforts so far in a new branch (`bugfix/spainhour/primal_numeric_type_fixes`). I'll try to fix this more comprehensively soon.